### PR TITLE
Blog App - processing data in models : Pull Request 

### PR DESCRIPTION
**Pull request by belaymit**
## Description

**_Now it's time to set up our models_**. To start with, we have tables for Users, Posts, Comments, and Likes, which means we need a model for each one. we've already set the foreign key in the table schema but be sure to write it here too!

## Project requirements

image
### Create models
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models
  - Remember that ``user_id`` column in ``posts`` table should be the ``foreign_key`` for the ``users`` table.
 
### Use models to insert data

- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:

```
first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') 
```
```second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')```

- Create at least 4 posts written by one of the users you created by running the following code:
``` first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post') ```

- Create at least 6 posts comments for one of the posts you created by running the following code:
```  Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' ) ```

- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

- Create custom methods
Your models should include some additional methods.

- Users
  - A method that returns the 3 most recent posts for a given user.
- Posts
  - A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.
  - Comments
  - A method that updates the comments counter for a post.
- Likes
  - A method that updates the likes counter for a post.
 
- Go to ``rails c`` and check if your methods are working.

### CODE REVIEW

File: app/models/blog_user.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class BlogUser < ApplicationRecord
-  has_many :posts
-  has_many :comments
+  has_many :posts, foreign_key: 'blog_user_id'
+  has_many :comments, foreign_key: 'blog_user_id'
+  has_many :likes, foreign_key: 'blog_user_id'
```
Comment (rloterh) [inline app/models/blog_user.rb]: - [ ] With the given [_ERD_](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), whiles establishing a _has many_ **association** for _User_ model, _author_id_ (_Author_ is aliasing _User_ model) should be referenced as **foreign key** in _Post_, _Comment,_ and _Like_ models. Kindly ensure to improve upon this aspect, and our code should be similar to:
```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

File: app/models/post.rb [hunk 2]
```diff
@@ -1,4 +1,17 @@
 class Post < ApplicationRecord
-  belongs_to :blog_user, foreign_key => 'blog_user_id'
+  belongs_to :blog_user
```
Comment (rloterh) [inline app/models/post.rb]: - [ ] As per the given _ERD_, establishing the _Post_ **association** should be such that _Post_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'`

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,4 +1,12 @@
 class Comment < ApplicationRecord
   belongs_to :blog_user
```
Comment (rloterh) [inline app/models/comment.rb]: - [ ] Similarly, establishing the _Comment_ **association** should be such that _Comment_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'`

File: app/models/like.rb [hunk 4]
```diff
@@ -1,4 +1,12 @@
 class Like < ApplicationRecord
   belongs_to :blog_user
```
Comment (rloterh) [inline app/models/like.rb]: - [ ] Again_, establishing the _Like_ **association** should be such that _Like_ belong to _User_ **through** _Author_, an alias of _User_, like below:
`belongs_to :author, class_name: 'User'`

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Belay, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦  🚗 _ 🏃 🌴  🏠 🌳 🏘️ 🏡  

## Highlights 👍  
- Good job so far with creating model associations!
- Professional ReadMe!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @belaymit , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**